course-material/interactive-queries-and-parameterization/introduction-to-ipywidgets.md

```diff
@@ -1,15 +1,376 @@
 ---
-jupyter:
-  jupytext:
-    text_representation:
-      extension: .md
-      format_name: markdown
-      format_version: '1.3'
-      jupytext_version: 1.14.5
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.14.6
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
 ---
 
-# Introduction to ipywidgets
+# Introduction to `ipywidgets`
 
+```{code-cell} ipython3
+:tags: [remove-cell]
 
+# this cell is hidden in the docs, only used to simulate
+# the getpass() call
+import ipywidgets as widgets
+from urllib.request import urlretrieve
+from zipfile import ZipFile
+import pandas as pd
+import os
+```
 
+Welcome back! We hope you gained a solid introduction to SQL and JupySQL in the first module.
 
+Before we begin, did you know that you can use widgets, eventful python objects that have a representation in the browser, to build fully interactive GUIs for your SQL query?
+
+In this section of Interactive Queries and Parameterization, we introduce `ipywidgets` and demonstrate how to create widgets and dynamically query a dataset in your workflow! Moreover, you can use widgets to synchronize stateful and stateless information between Python and JavaScript.
+
+`ipywidgets`, which is part of the Jupyter Widgets project, is a Python package that provides interactive HTML widgets for Jupyter notebooks and the IPython kernel. The package provides a basic, lightweight set of core form controls that use this framework. These controls comprise a text area, text box, select and multiselect controls, checkbox, sliders, tab panels, grid layout, etc.
```

Comment:
This paragraph's context is different. 

Ipywidgets have already been introduced, the focus is more on the interactive features of JupySQL and also to showcase how you can combine what the learner saw in the parameterize your queries and ipywidgets tutorials seen previously